client/state/comments/actions.js

```diff
@@ -0,0 +1,381 @@
+
+/***
+ * Internal handler for comments request success
+ * @param {Function} dispatch redux dispatch function
+ * @param {String} requestId request identifier
+ * @param {Number} siteId site identifier
+ * @param {Number} postId post identifier
+ * @param {Object[]} comments comments array
+ * @param {Number} totalCommentsCount comments count that the server have on that post
+ */
+function commentsRequestSuccess( dispatch, requestId, siteId, postId, comments, totalCommentsCount ) {
+	dispatch( {
+		type: COMMENTS_REQUEST_SUCCESS,
+		requestId: requestId
+	} );
+
+	dispatch( {
+		type: COMMENTS_RECEIVE,
+		siteId: siteId,
+		postId: postId,
+		comments: comments
+	} );
+
+	// if the api have returned comments count, dispatch it
+	if ( totalCommentsCount > -1 ) {
+		dispatch( {
+			type: COMMENTS_COUNT_RECEIVE,
+			siteId,
+			postId,
+			totalCommentsCount
+		} );
+	} else {
+		requestPostCommentsCount( siteId, postId )( dispatch );
+	}
+}
+
+/***
+ * Internal handler for comments request failure
+ * @param {Function} dispatch redux dispatch function
+ * @param {String} requestId request identifier
+ * @param {Object} err error object
+ */
+function commentsRequestFailure( dispatch, requestId, err ) {
+	dispatch( {
+		type: COMMENTS_REQUEST_FAILURE,
+		requestId: requestId,
+		error: err
+	} );
+
+	throw err;
+}
+
+/***
+ * Creates a thunk that requests comments for a given post
+ * @param {Number} siteId site identifier
+ * @param {Number} postId post identifier
+ * @returns {Function} thunk that requests comments for a given post
+ */
+export function requestPostComments( siteId, postId ) {
+	return ( dispatch, getState ) => {
+		const commentTargetId = createCommentTargetId( siteId, postId );
+		const currentCommentsState = getState().comments;
+
+		const earliestCommentDateForPost = currentCommentsState.earliestCommentDate.get( commentTargetId );
+
+		const query = {
+			order: 'DESC',
+			number: MAX_NUMBER_OF_COMMENTS_PER_FETCH
+		};
+
+		if ( earliestCommentDateForPost && earliestCommentDateForPost.toISOString ) {
+			query.before = earliestCommentDateForPost.toISOString();
+		}
+
+		const requestId = createRequestId( siteId, postId, query );
+
+		// if the request status is in-flight or completed successfully, no need to re-fetch it
+		if ( [ COMMENTS_REQUEST, COMMENTS_REQUEST_SUCCESS ].indexOf( currentCommentsState.queries.get( requestId ) ) !== -1 ) {
+			return;
+		}
+
+		dispatch( {
+			type: COMMENTS_REQUEST,
+			requestId: requestId
+		} );
+
+		// promise returned here is mainly for testing purposes
+		return wpcom.site( siteId )
+					.post( postId )
+					.comment()
+					.replies( query )
+					.then( ( { comments, found } ) => commentsRequestSuccess( dispatch, requestId, siteId, postId, comments, found ) )
+					.catch( ( err ) => commentsRequestFailure( dispatch, requestId, err ) );
+	};
+}
+
+
+let placeHolderCounter = 0;
```

Comment:
@gwwar Thanks for pointing that out, Since it's need to be unique only in the context of client, postId, siteId, [I've made it time based](https://github.com/Automattic/wp-calypso/pull/3143/files#diff-a1e5174c2173e0927694ec3bd26d79d4R144).